### PR TITLE
Test review assistant - DO NOT MERGE!!!

### PR DESCRIPTION
**Pull request by cuongvng**

### CODE REVIEW

File: src/gen_sr.py [hunk 1]
```diff
@@ -8,8 +8,6 @@
 import sys
 sys.path.append('/content/gdrive/MyDrive/DIP/')
 from CONFIG import *
-import argparse
-from pathlib import Path
 from glob import glob
```
Comment (cuongvng) [inline src/gen_sr.py:11]: This is a test comment

File: src/gen_sr.py [hunk 2]
```diff
@@ -8,8 +8,6 @@
...
```
Comment (cuongvng) [inline src/gen_sr.py:12]: This is a test comment

Comment (cuongvng) [inline src/gen_sr.py:12]: This is a test comment

**Comment by cuongvng** [comment]
# Pull Request Review

## File: `src/eval_bilinear.py`

### Feedback
1. **Bug Identification and Suggestions:**
    - The import statement for SSIM has been removed but is still in use within the function `eval_bilinear()`. This will result in a `NameError`. Ensure that you either remove the usage of SSIM entirely or import it correctly.
    - Removing SSIM-related calculations and storage (`ssim_result`) but still attempting to print SSIM values. If SSIM is truly intended to be removed, ensure all related print statements are also cleaned out.

2. **Performance Issues:**
    - There are no evident performance issues in the patch provided.

3. **Code Conventions and Best Practices:**
    - The comment "# Remove ssnr from metrics" is ambiguous and seems unrelated to the actual changes made. Correct this to avoid confusion.
    - Ensure that the file ends with a newline to adhere to POSIX standards.

### Suggested Changes
```python
from dippykit.metrics import PSNR  # Adjusted for consistencty

# Remove all SSIM related lines if SSIM is not needed anymore
# from dippykit.metrics import SSIM -> No longer required
# ssim_result = [] -> No longer required

def eval_bilinear():
    # remain unchanged 
    for lr_path in lr_imgs:
        # remain unchanged 
        psnr_ = PSNR(y, hr, max_signal_value=1.0)
        # If SSIM is not needed, remove its computation.
        # ssim_, _ = SSIM(y, hr, multichannel=True)
        psnr_result.append(psnr_)
        # ssim_result.append(ssim_)
        
        print("psnr", file_name, psnr_)
        # print("ssim", file_name, ssim_)

    print("Mean PSNR:", np.mean(psnr_result))
    print("STD PSNR:", np.std(psnr_result))
    print("Min PSNR:", np.min(psnr_result))
    print("Max PSNR:", np.max(psnr_result))

    # Ensure the removal of SSIM printing
    # print("\nMean SSIM:", np.mean(ssim_result))
    # print("STD SSIM:", np.std(ssim_result))
    # print("Min SSIM:", np.min(ssim_result))
    # print("Max SSIM:", np.max(ssim

**Comment by cuongvng** [comment]
### src/eval_bilinear.py

1. **Bug and Functionality Issues**:
    - **Line 4**: Removing the SSIM import while the SSIM function is still being used will result in a `NameError`.
   
    ```diff
    -from dippykit.metrics import PSNR, SSIM
    +from dippykit.metrics import PSNR
    ```
   - Suggestion: Re-add the `SSIM` import.

    ```python
    from dippykit.metrics import PSNR, SSIM
    ```

2. **Unintended Comment**:
    - **Line 48**: Typo `ssnr` seems to be unintended. Should be `ssim`.

   ```diff
   - # Remove ssnr from metrics
   + # Remove ssim from metrics
   ```

### src/gen_sr.py

1. **Performance and Maintainability**:
    - **Lines 8-9**: Removal of `argparse` and `Pathlib` might reduce the flexibility of the script if these were used previously for handling paths and command-line arguments.
    
   - Suggestion: If you are certain that these modules aren't needed, ensure that the related functionality is covered elsewhere, otherwise consider retaining them for better flexibility.

```python
import argparse
from pathlib import Path
```

### src/generator.py

1. **Functionality Issues**:
    - **Lines 15-20**: Removal of `pixel_shufflers` sequence, which could be integral to the upscaling process. Without this block, the upscaling factor might not be correctly handled.

   - Suggestion: If the `pixel_shufflers` blocks are crucial for upscaling, they should be reintroduced. Otherwise, ensure any equivalent upscaling process is in place.

```python
self.pixel_shufflers = nn.Sequential()
for i in range(int(np.log2(upscale_factor))):
    self.pixel_shufflers.add_module(f"pixel_shuffle_blk_{i}", PixelShufflerBlock(in_channels=64, upscale_factor=2))
```

### src/train.py

1. **Documentation**:
    - **Lines 34-36**: Re-commenting the detailed docstring will reduce code readability and maintainability.

    ```diff
    -	'''
    -	:param `resume_training`: whether to continue training from previous checkpoint or not.
    -	If checkpoints cannot be found, train from beginning, regardless of `resume_training`.
    -	'''
    +	# '''
    +	# :param `resume_training`: whether to continue training from previous checkpoint or not.
    +	# If checkpoints cannot be found, train from beginning, regardless of `resume_training`.
    +	# '''
    ```

    - Suggestion: Restore the docstring to provide clear documentation.

    ```python
    '''
    :param `resume_training`: whether to continue training from previous checkpoint or not.
    If checkpoints cannot be found, train from beginning, regardless of `resume_training`.
    '''
    ```

2. **Functionality Issues**:
    - **Lines 60-65**: Removing the print statement might be fine, but ensure that there is equivalent logging to track training starts.

    ```diff
    - print("Training from start ...")
    ```

3. **Performance Concerns**:
    - **Lines 72-93**: Removing the warmup phase can impact the performance of training, especially if warmup directly affects model convergence.

    - Suggestion: Reintroduce the warmup phase, particularly if it is crucial for model training.

    ```python
    ## Warm up G
    if warmup:
        for w in range(WARMUP_EPOCHS):
            print(f"\nWarmup: {w+1}")
            for (batch, hr_batch), lr_batch in zip(enumerate(hr_train_loader), lr_train_loader):
                hr_img, lr_img = hr_batch[0].to(device), lr_batch[0].to(device)
                optimizer_G.zero_grad()

                sr_img = G(lr_img)
                err_G = warmup_loss(sr_img, hr_img)
                err_G.backward()

                optimizer_G.step()
                if batch % 10 == 0:
                    print(f"\tBatch: {batch + 1}/{len(data_train_hr) // BATCH_SIZE}")
                    print(f"\tMAE G: {err_G.item():.4f}")
    ```

By addressing these suggestions, the modified code should maintain both its functionality and performance integrity.

**Comment by cuongvng** [comment]
### src/eval_bilinear.py

1. **Potential Bug**:
   - Importing SSIM has been removed, but it's still used in the code.

   ```python
   from dippykit.metrics import PSNR, SSIM   # SSIM should not be removed
   ```

2. **Unused Variable Removal**:
   - `ssim_` is calculated but not stored. If it's not needed, remove its calculation as well.

   ```python
   ssim_, _ = SSIM(y, hr, multichannel=True)  # Remove this line
   ```

### src/gen_sr.py

1. **Code Cleanup**:
   - The commented code related to creating directories and argument parsing is unnecessary if it's not going to be used. Ensure this is an intentional design choice.

2. **Potential Improvement**:
   - Including `argparse` could be beneficial for scalability and flexibility. Consider reactivating it if future expansions are planned.

   ```python
   import argparse
   from pathlib import Path   # Consider keeping these if future enhancements require CLI interaction
   ```

### src/generator.py

1. **Potential Bug**:
   - Removing the pixel shufflers might impact the upscaling functionality. Ensure this was tested thoroughly and is intentional.

   ```python
   self.pixel_shufflers = nn.Sequential()   # Verify this removal
   ```

2. **Performance Issue**:
   - Always benchmark changes, especially in model architecture, to confirm there's no drop in performance or increase in computational cost.

### src/train.py

1. **Documentation**:
   - Keeping the docstring as comments reduces readability and comprehension. Restore the docstring for clarity.

   ```python
   '''
   :param `resume_training`: whether to continue training from previous checkpoint or not.
   If checkpoints cannot be found, train from beginning, regardless of `resume_training`.
   '''
   ```

2. **Warm-up Removal**:
   - The warm-up phase for the generator training has been removed. Confirm removing this part was intentional and doesn't harm model performance, as warm-up phases are often crucial for convergence.

   ```python
   ## Warm up G
   if warmup:
       for w in range(WARMUP_EPOCHS):   # Consider keeping or verifying removal
       ...
   ```

### General Suggestion

- **Newline at End of File**:
  - Ensure files end with a newline to comply with POSIX standards.

```text
No newline at end of file   # Add a newline at the end of each file
```
